### ISSUE TITLE
"Too many values to unpack" when using KS4 version 4.0.11

### ISSUE THREAD

**Issue by EricKenjiLee** [SpikeInterface/spikeinterface#3007]
I updated to the newest release of Kilosort4, version 4.0.11, and it looks like this has now broken compatibility in SpikeInterface (version 0.100.6). It looks like the number of outputs in the new version of the sorter does not match previously. Seems the culprits are the new "shift" and "scale" parameters (https://github.com/MouseLand/Kilosort/releases/tag/v4.0.11). The particular section of Kilosort's code is [here](https://github.com/MouseLand/Kilosort/blob/221ee2641d29827569734aa3af60b90c7c282190/kilosort/run_kilosort.py#L334).

```
Error running kilosort4

---------------------------------------------------------------------------
SpikeSortingError                         Traceback (most recent call last)
Cell In[21], line 1
----> 1 sorting_ks4_DREDge = si.run_sorter('kilosort4', rec_DREDge_drift, output_folder=base_folder / 'kilosort4_DREDge_output',
      2                         docker_image=False, verbose=True, **params_kilosort4)

File ~/anaconda3/envs/si_env/lib/python3.11/site-packages/spikeinterface/sorters/runsorter.py:175, in run_sorter(sorter_name, recording, output_folder, remove_existing_folder, delete_output_folder, verbose, raise_error, docker_image, singularity_image, delete_container_files, with_output, **sorter_params)
    168             container_image = singularity_image
    169     return run_sorter_container(
    170         container_image=container_image,
    171         mode=mode,
    172         **common_kwargs,
    173     )
--> 175 return run_sorter_local(**common_kwargs)

File ~/anaconda3/envs/si_env/lib/python3.11/site-packages/spikeinterface/sorters/runsorter.py:225, in run_sorter_local(sorter_name, recording, output_folder, remove_existing_folder, delete_output_folder, verbose, raise_error, with_output, **sorter_params)
    223 SorterClass.set_params_to_folder(recording, output_folder, sorter_params, verbose)
    224 SorterClass.setup_recording(recording, output_folder, verbose=verbose)
--> 225 SorterClass.run_from_folder(output_folder, raise_error, verbose)
    226 if with_output:
    227     sorting = SorterClass.get_result_from_folder(output_folder, register_recording=True, sorting_info=True)

File ~/anaconda3/envs/si_env/lib/python3.11/site-packages/spikeinterface/sorters/basesorter.py:293, in BaseSorter.run_from_folder(cls, output_folder, raise_error, verbose)
    290         print(f"{sorter_name} run time {run_time:0.2f}s")
    292 if has_error and raise_error:
--> 293     raise SpikeSortingError(
    294         f"Spike sorting error trace:\n{log['error_trace']}\n"
    295         f"Spike sorting failed. You can inspect the runtime trace in {output_folder}/spikeinterface_log.json."
    296     )
    298 return run_time

SpikeSortingError: Spike sorting error trace:
Traceback (most recent call last):
  File "/home/chandlab/anaconda3/envs/si_env/lib/python3.11/site-packages/spikeinterface/sorters/basesorter.py", line 258, in run_from_folder
    SorterClass._run_from_folder(sorter_output_folder, sorter_params, verbose)
  File "/home/chandlab/anaconda3/envs/si_env/lib/python3.11/site-packages/spikeinterface/sorters/external/kilosort4.py", line 196, in _run_from_folder
    n_chan_bin, fs, NT, nt, twav_min, chan_map, dtype, do_CAR, invert, _, _, tmin, tmax, artifact = (
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: too many values to unpack (expected 14)

Spike sorting failed. You can inspect the runtime trace in /home/chandlab/HDD1/NPixExps/kilosort4_DREDge_output/spikeinterface_log.json.
```

**Comment by alejoe91** [comment]
Hi @EricKenjiLee 

Kilosort4 is moving at a fast pace and we're lagging behind in properly supporting all of its versions.

Can you try to install spike interface 0.100.7 (released on Friday) and kilosort4 4.0.6?

That is the last combination that is supported. For any more recent version, we cannot guarantee for now that it will work as expected.

**Comment by EricKenjiLee** [comment]
Okay yeah thank you! Totally understood; just wanted you all to be aware!

**Comment by alejoe91** [comment]
Thanks! We'll keep this open.

Let us know if it works!

**Comment by EricKenjiLee** [comment]
0.100.7 worked perfectly with 4.0.6! Sorry for opening all the issues!

**Comment by FrancescoNegri** [comment]
I do not know what kind of effort it takes to support newer Kilosort 4 versions. However, releases previous to 4.0.7 present a considerable issue regarding the first batch of sorted spikes. As a matter of fact, there is a misalignment between the spike timestamps and the actual data. The related Kilosort repo issue is MouseLand/Kilosort#673.

**Comment by alejoe91** [comment]
Fixed by #3055
